### PR TITLE
CP-11278: Add supported nfs versions to sr-probe

### PR DESCRIPTION
**Pull request by siddharthv**
This patch adds the list of supported nfs versions for a given server
based on the 'nfsversion' key in device_config.

Signed-off-by: Siddharth Vinothkumar <siddharth.vinothkumar@citrix.com>

### CODE REVIEW

**Comment by siddharthv** [comment]
@chandrikas I think we should use a new key in device_config, perhaps `nfsprobe` rather than relying on `nfsversion`.

**Comment by chandrikas** [comment]
Yes, just a flag to indicate we want to probe for supported version. Why can't we return supported version by default BTW? What is the use case for that?

File: drivers/NFSSR.py [hunk 1]
```diff
@@ -110,7 +110,10 @@ def validate_remotepath(self, scan):
 
     def check_server(self):
         try:
-            nfs.check_server_tcp(self.remoteserver, self.nfsversion)
+            if self.srcmd.params.get('device_config').has_key(PROBEVERSION):
```
Comment (chandrikas) [inline drivers/NFSSR.py]: Can we check self.dconf? That should already be populated with all device_config parameters.

File: drivers/NFSSR.py [hunk 2]
```diff
@@ -118,9 +121,14 @@ def check_server(self):
 
     def mount(self, mountpoint, remotepath, timeout = 0):
         try:
-            nfs.soft_mount(
-                mountpoint, self.remoteserver, remotepath, self.transport,
-                timeout=timeout, nfsversion=self.nfsversion)
+            if self.srcmd.params.get('device_config').has_key(PROBEVERSION):
```
Comment (chandrikas) [inline drivers/NFSSR.py]: Can we check self.dconf? That should already be populated with all device_config parameters.

File: drivers/NFSSR.py [hunk 3]
```diff
@@ -151,7 +159,7 @@ def probe(self):
 
         self.mount(temppath, self.remotepath)
         try:
-            return nfs.scan_srlist(temppath)
+            return nfs.scan_srlist(temppath, self.dconf.get('server'), self.srcmd.params.get('device_config'))
```
Comment (chandrikas) [inline drivers/NFSSR.py]: We could just pass self.dconf to nfs.scan_srlist here. That has all the parameters we need.

**Comment by chandrikas** [comment]
@siddharthv , the patch looks good. Please change the commit message to reflect the device_config key we use to probe version (commit message still says nfsversion). Changes can be pushed once commit message is changed.

**Pull request closed without merging**